### PR TITLE
Correct df7_10 road alignments according the changes to the EEA template

### PR DESCRIPTION
**Pull request by annat2022**
SVC-1872

### CODE REVIEW

File: Snippets/end.Checks.groovy [hunk 1]
```diff
@@ -467,10 +467,11 @@ def existsCheck(String codeMissing, String sheet, String field, boolean isWarnin
     generalCheck(null, codeMissing, sheet, field, null, exists(), null, isWarning)
 }
 
-def urlCheck(String codeInvalid, String codeMissing, String sheet, String field, String mustStartWith = null, boolean isWarning = false) {
-    generalCheck(codeInvalid, codeMissing, sheet, field, 'is not a valid URL',
+def urlCheck(String codeInvalid, String codeMissing, String sheet, String field, String mustStartWith = null, boolean isWarning = false, long maxLength = _project.vars.MAX_URL_LENGTH as long) {
+    (generalCheck(codeInvalid, codeMissing, sheet, field, 'is not a valid URL',
```
Comment (florianesser) [inline Snippets/end.Checks.groovy]: The outer brackets around the whole statement can be removed

File: Validation/MajorRoads/UBA-DE_HVS_to_END_DF7_10_Roads.halex.alignment.xml [hunk 2]
```diff
@@ -48,30 +33,80 @@ withTransformationContext {&#13;
...
-		limitValues_otherCriteriaDescription( c.Allgemeines.infoGrenzwert.value() )&#13;
-		limitValues_noiseLimitReportIdIdentifier( 'LR_DE_00_1' )&#13;
+	def ags_rs = c.Allgemeines.ags_rs.value()&#13;
+		&#13;
+	switch (ags_rs.length()) {&#13;
+		case 8:		&#13;
+			// AGS of Gemeinde is equivalent to LAU code	&#13;
+			_target {&#13;
+				actionPlanId_identifier( c.Allgemeines.kennungPlan.value() )&#13;
+				allInLAUCode( 'yes' )&#13;
```
Comment (florianesser) [inline Validation/MajorRoads/UBA-DE_HVS_to_END_DF7_10_Roads.halex.alignment.xml]: The value here should be `Yes` (see `CODELIST_Boolean()` helper)

File: Validation/MajorRoads/UBA-DE_HVS_to_END_DF7_10_Roads.halex.alignment.xml [hunk 3]
```diff
@@ -48,30 +33,80 @@ withTransformationContext {&#13;
...
+				LAUCode( ags_rs )&#13;
+			}&#13;
+			break&#13;
+		case 9:&#13;
+			// get LAU codes for RS of Amt (MV) or Verbandsgemeinde like classified in lookup-table 'VWG-RS_zu_LAUCodes'&#13;
+			def lau = _snippets.helperFunctions.getLookupTableValue(ags_rs, 'b5e5b305-6d74-4790-89a7-3a222924e400')&#13;
+			_target {&#13;
+				actionPlanId_identifier( c.Allgemeines.kennungPlan.value() )&#13;
+				allInLAUCode( 'yes' )&#13;
```
Comment (florianesser) [inline Validation/MajorRoads/UBA-DE_HVS_to_END_DF7_10_Roads.halex.alignment.xml]: `Yes`

File: Validation/MajorRoads/UBA-DE_HVS_to_END_DF7_10_Roads.halex.alignment.xml [hunk 4]
```diff
@@ -48,30 +33,80 @@ withTransformationContext {&#13;
...
+withTransformationContext {&#13;
+	def c = _.context.collector(it)&#13;
+	&#13;
+	def ags_rs = c.Allgemeines.ags_rs.value()&#13;
+	if (ags_rs.length() == 3) {		&#13;
+	// get NUTS code for RS of Regierungsbezirk like classified in lookup-table 'RB-RS_zu_NUTSCode'&#13;
+	nuts = _snippets.helperFunctions.getLookupTableValue(ags_rs, 'ce7f4ce5-ceb6-47ef-bb9d-b79c930792c4')&#13;
+		_target {&#13;
+			actionPlanId_identifier( c.Allgemeines.kennungPlan.value() )&#13;
+			allInNUTSCode( 'yes' )&#13;
```
Comment (florianesser) [inline Validation/MajorRoads/UBA-DE_HVS_to_END_DF7_10_Roads.halex.alignment.xml]: `Yes`

File: Validation/MajorRoads/UBA-DE_HVS_to_END_DF7_10_Roads.halex.alignment.xml [hunk 5]
```diff
@@ -48,30 +33,80 @@ withTransformationContext {&#13;
...
+        </target>
+        <complexParameter name="script">
+            <core:text xmlns:core="http://www.esdi-humboldt.eu/hale/core" xml:space="preserve">
+withTransformationContext {&#13;
+	def c = _.context.collector(it)&#13;
+	&#13;
+	def ags_rs = c.Allgemeines.ags_rs.value()&#13;
+	if (ags_rs.length() == 3) {		&#13;
+	// get NUTS code for RS of Regierungsbezirk like classified in lookup-table 'RB-RS_zu_NUTSCode'&#13;
+	nuts = _snippets.helperFunctions.getLookupTableValue(ags_rs, 'ce7f4ce5-ceb6-47ef-bb9d-b79c930792c4')&#13;
```
Comment (florianesser) [inline Validation/MajorRoads/UBA-DE_HVS_to_END_DF7_10_Roads.halex.alignment.xml]: Indentation looks wrong

**Review by florianesser — CHANGES_REQUESTED** [review]
Generally LGTM, just noticed a couple of smaller things to fix before merging.

Also, the first line of the commit message ` feat(df7_10): implement check for max URL length` starts with a space, that's why the commit message check failed.

File: Validation/MajorRoads/UBA-DE_HVS_to_END_DF7_10_Roads.halex.alignment.xml (hunk 2 above)
Comment (annat2022) [reply Validation/MajorRoads/UBA-DE_HVS_to_END_DF7_10_Roads.halex.alignment.xml]: Good catch, thanks for the hint. I corrected it now, also for the two other cases

File: Snippets/end.Checks.groovy (hunk 1 above)
Comment (annat2022) [reply Snippets/end.Checks.groovy]: I removed them.

File: Validation/MajorRoads/UBA-DE_HVS_to_END_DF7_10_Roads.halex.alignment.xml (hunk 5 above)
Comment (annat2022) [reply Validation/MajorRoads/UBA-DE_HVS_to_END_DF7_10_Roads.halex.alignment.xml]: Corrected it now.

**Comment by annat2022** [comment]
> Also, the first line of the commit message  feat(df7_10): implement check for max URL length starts with a space, that's why the commit message check failed.

Thanks for the hint. I was already wondering why it is failing but missed that space...

**Review by florianesser — APPROVED** [review]
LGTM

**Pull request merged**